### DISCUSSION TITLE
Can a deploy key create a pull request?

### DISCUSSION THREAD

**Discussion by 1f604** [community/community#21997] [Pull Requests]
<p>Let’s say I created a deploy key with write permissions to repository A.</p>
<p>Let’s say repository A has a protected master branch that has " Require a pull request before merging " enabled and “Allow force pushes” and “Allow deletions” disabled for everyone including admins.</p>
<p>If a malicious attacker steals that deploy key, what damage can they do to the repository? Can they create a pull request? Can they merge that pull request?</p>
<p>Is my understanding correct that the only thing they can do is to create, modify, or delete unprotected branches in that repository? They can’t affect the protected branches in any way?</p>

**Comment by mpboom** [comment, marked as answer]
<p>Hi <a class="mention" href="/1f604">@1f604</a> and welcome to the forum! Yes, I believe that is correct. They cannot execute GitHub related actions like creating/updating PRs/Issues or get access to your GitHub account in any way.</p>
<p>Keep in mind you can also create a deploy key that can just pull if you’re worried about changes being pushed.</p>
